config/README.md

```diff
@@ -54,6 +106,25 @@ At this point the script(s) in this repository will work. However, if we wish to
 ```
 CREATE INDEX idx_publication_date ON uspto_patents (publication_date);
 CREATE INDEX idx_publication_title ON uspto_patents ((lower(publication_title)));
+
+CREATE INDEX IF NOT EXISTS uspto_publication_number ON uspto_referential_documents (uspto_publication_number);
+
+DO $$
+BEGIN
+    IF (select Substr(setting, 1, strpos(setting, '.')-1) from pg_settings where name = 'server_version')::INTEGER = 15 THEN
+        --FOR PSQL >= 15 which allows null not distinct
+        --CREATE UNIQUE INDEX IF NOT EXISTS patent_reference_constraint on uspto_referential_documents
+        --    (uspto_publication_number, reference, document_type, country, (metadata->>'kind')) NULLS NOT DISTINCT;
+        RAISE WARNING "UNCOMMENT THIS SECTION FOR PSQL>=15; NO INDEXES APPLIED"
+    ELSE
+        --FOR PSQL < 15
+        CREATE UNIQUE INDEX IF NOT EXISTS patent_reference_constraint_null on uspto_referential_documents
+            (uspto_publication_number, COALESCE(reference, ''), document_type, COALESCE(country, '')) WHERE (metadata->>'kind') is NULL;
+        CREATE UNIQUE INDEX IF NOT EXISTS patent_reference_constraint_not_null on uspto_referential_documents
+            (uspto_publication_number, COALESCE(reference, ''), document_type, COALESCE(country, ''), (metadata->>'kind')) WHERE (metadata->>'kind') is NOT NULL;
```

Comment:
I'd prefer if we don't create a unique index on a JSON; it creates a very messy index that will be slower and harder to maintain

Reply:
fixed